src/System.ComponentModel.Composition/tests/System/ComponentModel/Composition/CompositionContainerTests.cs

```diff
@@ -3022,6 +3023,33 @@ public void TestExportedValueCachesNullValue()
             Assert.Null(container.GetExportedValue<string>("Property"));
         }
 
+        [Fact]
+        public void TestExportedValueUsingWhereClause_ExportSuccessful()
+        {
+            CompositionContainer container = new CompositionContainer(new AssemblyCatalog(typeof(AssemblyCatalogTestsReflectionContext).Assembly));
+            IMefCollection<DerivedClass, BaseClass> i = container.GetExportedValue<IMefCollection<DerivedClass, BaseClass>>("UsingWhereClause");
+            Assert.NotNull(i);
+        }
+
+        public interface IMefCollection<TC, TP> : IList<TC>, IMefCollection where TC : TP { }
+        public class BaseClass { }
+        public class DerivedClass : BaseClass { }
+
+        public interface IMefCollection
+        {
+            void WriteTypeNameOfSecondParam();
+        }
+
+        [Export("UsingWhereClause", typeof(IMefCollection<,>))]
+        public class MefCollection<TC, TP> : ObservableCollection<TC>, IMefCollection<TC, TP> where TC : TP
+        {
+            public void WriteTypeNameOfSecondParam()
+            {
+                Console.WriteLine("TC Type: " + typeof(TC).ToString());
```

Comment:
We usually don't write to the console's output during tests. Is this needed?

Reply:
No you're right. I better remove. Thanks